README.md

```diff
@@ -238,7 +238,12 @@ Creates and returns an instance of an IPFS node. Use the `options` argument to s
 
 - `config` (object) Modify the default IPFS node config. Find the Node.js defaults at [`src/core/runtime/config-nodejs.js`](https://github.com/ipfs/js-ipfs/tree/master/src/core/runtime/config-nodejs.js) and the browser defaults at [`src/core/runtime/config-browser.js`](https://github.com/ipfs/js-ipfs/tree/master/src/core/runtime/config-browser.js). This object will be *merged* with the default config; it will not replace it.
 
-- `libp2p` (object) add custom modules to the libp2p stack of your node
+- `libp2p` (object or function(ipfs, config)) add custom modules to the libp2p stack of your node
+
+    The libp2p option allows you to build your libp2p node by configuration, or via a generator. If you are looking to just modify the below options, using the object format is the quickest way to get the default features of libp2p. If you need to create a more customized libp2p node, such as with custom transports or peer/content routers that need some of the ipfs data on startup, a generator is a great way to achieve this.
```

Comment:
I'm wondering if "generator" might be confused with an actual JS [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*)? IMHO it would be more appropriate to call it a [factory function](https://en.wikipedia.org/wiki/Factory_method_pattern).

Reply:
Yes I agree that could get confusing, I will update the references to *factory*.

Reply:
I think this happened but then got changed back?

Reply:
wow, I spaced on that one. Came back from vacation saw some inconsistencies between factory/generator and changed everything the wrong way. Everything should be factory now, aside from the branch name.